app/api/views/brief_response_contacts.py

```diff
@@ -0,0 +1,120 @@
+from flask import jsonify
+from flask_login import current_user, login_required
+from app.api import api
+from app.api.helpers import not_found
+from ...models import AuditEvent
+from app.api.services import brief_response_contact_service, audit_service, audit_types
+from app.api.helpers import role_required
+from ...utils import get_json_from_request
+import rollbar
+
+
+@api.route('/brief-response-contact/<int:brief_id>', methods=['PUT'])
+@login_required
+@role_required('supplier')
+def save_brief_response_contact(brief_id):
+    """Save brief response contact (role=supplier)
+    ---
+    tags:
+      - "Brief Response Contact"
+    security:
+      - basicAuth: []
+    parameters:
+      - name: brief_id
+        in: path
+        type: number
+        required: true
+      - name: body
+        in: body
+        required: true
+        schema:
+          id: BriefResponseContactSave
+          required:
+            - emailAddress
+          properties:
+            emailAddress:
+              type: string
+    responses:
+      200:
+        description: Successfully updated a candidate
+        schema:
+          id: BriefResponseContact
+      400:
+        description: brief_id not found
+    """
+    brief_response_contact = (brief_response_contact_service
+                              .find(brief_id=brief_id,
+                                    supplier_code=current_user.supplier_code)
+                              .one_or_none())
+
+    if brief_response_contact:
+        brief_response_contact_json = get_json_from_request()
+        brief_response_contact.email_address = brief_response_contact_json['emailAddress']
+        brief_response_contact_service.save(brief_response_contact)
+
+        try:
+            audit = AuditEvent(
```

Comment:
idea: have the try/catch in the audit service - its a lot of code to keep copy/pasting

Reply:
fixed in latest commit